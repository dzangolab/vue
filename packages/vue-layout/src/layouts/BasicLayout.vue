<template>
  <div class="layout basic">
    <slot name="header">
      <AppHeader :no-locale-switcher="noLocaleSwitcher">
        <template #logo>
          <slot name="logo" />
        </template>
        <template #menu>
          <slot name="menu" />
        </template>
        <template #userMenu>
          <slot name="userMenu" />
        </template>
        <template #locales>
          <slot name="locales" />
        </template>
      </AppHeader>
    </slot>

    <main>
      <slot name="default"></slot>
    </main>

    <slot name="footer">
      <AppFooter>
        <template #copyright>
          <slot name="copyright" />
        </template>
        <template #social>
          <slot name="social" />
        </template>
        <template #version>
          <slot name="version" />
        </template>
      </AppFooter>
    </slot>
  </div>
</template>

<script lang="ts">
export default {
  name: "BasicLayout",
};
</script>

<script setup lang="ts">
import AppFooter from "../components/AppFooter.vue";
import AppHeader from "../components/AppHeader.vue";

defineProps({
  noLocaleSwitcher: Boolean,
});
</script>
