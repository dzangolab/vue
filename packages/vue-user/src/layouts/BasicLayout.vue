<template>
  <OriginalBasicLayout>
    <template #header>
      <AppHeader>
        <template #logo>
          <Logo :route="home" />
        </template>
        <template #userMenu>
          <UserMenu class="userMenu" />
        </template>
      </AppHeader>
    </template>

    <slot></slot>
  </OriginalBasicLayout>
</template>

<script lang="ts">
export default {
  name: "BasicLayout",
};
</script>

<script setup lang="ts">
import { useConfig } from "@dzangolab/vue3-config";
import {
  AppHeader,
  BasicLayout as OriginalBasicLayout,
  Logo,
} from "@dzangolab/vue3-layout";
import { storeToRefs } from "pinia";
import { computed } from "vue";

import UserMenu from "../components/UserMenu.vue";
import useUserStore from "../store";

const userStore = useUserStore();

const { user } = storeToRefs(userStore);

const home = computed(() => {
  const { layout: layoutConfig, user: userConfig } = useConfig();

  return user.value
    ? userConfig && userConfig?.routes?.home
      ? userConfig.routes.home
      : "profile"
    : layoutConfig && layoutConfig?.homeRoute
    ? layoutConfig.homeRoute
    : undefined;
});
</script>
