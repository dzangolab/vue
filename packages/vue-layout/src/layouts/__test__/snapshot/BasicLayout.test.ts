import { shallowMount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import BasicLayout from "../../BasicLayout.vue";

import type { VueWrapper } from "@vue/test-utils";

describe("BasicLayout", () => {
  it("matches snapshot", () => {
    const wrapper: VueWrapper = shallowMount(BasicLayout);

    expect(wrapper.element).toMatchSnapshot();
  });
});
