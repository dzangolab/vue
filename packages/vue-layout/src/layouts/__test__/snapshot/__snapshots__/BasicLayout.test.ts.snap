// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BasicLayout > matches snapshot 1`] = `
<div
  class="basic-layout"
>
  
  <app-header-stub />
  
  <main>
    
    
  </main>
  
  <app-footer-stub />
  
</div>
`;
